provisioning/device/samples/register_symkey.js

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var iotHubTransport = require('azure-iot-device-mqtt').Mqtt;
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+
+var ProvisioningTransport = require('azure-iot-provisioning-device-http').Http;
+// Feel free to change the preceding using statement to anyone of the following if you would like to try another protocol.
+// var ProvisioningTransport = require('azure-iot-provisioning-device-amqp').Amqp;
+// var ProvisioningTransport = require('azure-iot-provisioning-device-amqp').AmqpWs;
+// var ProvisioningTransport = require('azure-iot-provisioning-device-mqtt').Mqtt;
+// var ProvisioningTransport = require('azure-iot-provisioning-device-mqtt').MqttWs;
+
+var symmetricKeySecurity = require('azure-iot-security-symmetric-key');
+var ProvisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
+
+//
+// For the public clouds the address of the provisioning host would be: global.azure-devices-provisioning.net
+//
+var provisioningHost = '<the host for provisioning>';
+
+//
+// You can find your idScope in the portal overview section for your dps instance.
+//
+var idScope = '<id scope for dps instance>';
+
+//
+// The registration id of the device to be registered.
+//
+var registrationId = '<registration id for device>';
+
+var symmetricKey = '<symmetric key for device>';
```

Comment:
>'<symmetric key for device>' [](start = 19, length = 28)

let's encourage the use of env variables here instead of inserting the string in code.

Reply:
Dony

---
In reply to: [231344818](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231344818) [](ancestors = 231344818)